includes/admin/donors/class-donor-table.php

```diff
@@ -68,36 +68,86 @@ public function __construct() {
 	 * @return void
 	 */
-	public function search_box( $text, $input_id ) {
-		$input_id = $input_id . '-search-input';
-
-		if ( ! empty( $_REQUEST['orderby'] ) ) {
-			echo sprintf( '<input type="hidden" name="orderby" value="%1$s" />', esc_attr( $_REQUEST['orderby'] ) );
-		}
+	public function advanced_filters() {
+		$start_date  = isset( $_GET['start-date'] ) ? give_clean( $_GET['start-date'] ) : null;
+		$end_date    = isset( $_GET['end-date'] ) ? give_clean( $_GET['end-date'] ) : null;
+		$status      = isset( $_GET['status'] ) ? give_clean( $_GET['status'] ) : '';
+		$donor       = isset( $_GET['donor'] ) ? absint( $_GET['donor'] ) : '';
+		$search      = isset( $_GET['donor_search_input'] ) ? give_clean( $_GET['donor_search_input'] ) : '';
+		$form_id     = ! empty( $_GET['form_id'] ) ? absint( $_GET['form_id'] ) : 0;
 
-		if ( ! empty( $_REQUEST['order'] ) ) {
-			echo sprintf( '<input type="hidden" name="order" value="%1$s" />', esc_attr( $_REQUEST['order'] ) );
-		}
 		?>
-		<p class="search-box" role="search">
-			<label class="screen-reader-text" for="<?php echo $input_id ?>"><?php echo $text; ?>:</label>
-			<input type="search" id="<?php echo $input_id ?>" name="s" value="<?php _admin_search_query(); ?>"/>
-			<?php submit_button( $text, 'button', false, false, array(
-				'ID' => 'search-submit',
-			) ); ?>
-		</p>
+		<div id="give-donor-filters" class="give-filters">
+			<div class="donor_search_box">
+				<input type="text" id="give-donors-search-input" placeholder="<?php _e( 'Search Donors', 'give' ); ?>" name="donor_search_input" />
+			</div>
+			<div id="give-donor-date-filters">
+				<div class="give-filter give-filter-half">
+					<label for="start-date"
+					       class="give-start-date-label"><?php _e( 'Start Date', 'give' ); ?></label>
+					<input type="text" id="start-date" name="start-date" class="give_datepicker" autocomplete="off"
+					       value="<?php printf( esc_attr( $start_date ) ); ?>" placeholder="<?php _e( 'Start Date', 'give' ); ?>" />
+				</div>
+				<div class="give-filter give-filter-half">
+					<label for="end-date" class="give-end-date-label"><?php _e( 'End Date', 'give' ); ?></label>
+					<input type="text" id="end-date" name="end-date" class="give_datepicker" autocomplete="off"
+					       value="<?php printf( esc_attr( $end_date ) ); ?>" placeholder="<?php _e( 'End Date', 'give' ); ?>" />
+				</div>
+			</div>
+			<div id="give-payment-form-filter" class="give-filter">
+				<label for="give-donation-forms-filter"
+				       class="give-donation-forms-filter-label"><?php _e( 'Form', 'give' ); ?></label>
+				<?php
+				// Filter Donations by Donation Forms.
+				echo Give()->html->forms_dropdown(
+					array(
+						'name'     => 'form_id',
+						'id'       => 'give-donation-forms-filter',
+						'class'    => 'give-donation-forms-filter',
+						'selected' => $form_id, // Make sure to have $form_id set to 0, if there is no selection.
+						'chosen'   => true,
+						'number'   => - 1,
```

Comment:
Don't pull all donation forms. Use AJAX chosen2 dropdown and pull latest 30.

Reply:
If pulling all donation forms on sites with 1000s of forms it would be a very long query...